### ISSUE TITLE
src/xmoto/LuaLibBase.cpp uses deprecated lua functions

### ISSUE THREAD

**Issue by winterheart** [xmoto/xmoto#95]
src/xmoto/LuaLibBase.cpp uses deprecated lua 5.0 function luaL_openlib: https://github.com/xmoto/xmoto/blob/master/src/xmoto/LuaLibBase.cpp#L45, which produces compilation error if lua 5.1 stripped down from deprecation functions:

```
/var/tmp/portage/games-sports/xmoto-0.6.0/work/xmoto-0.6.0/src/xmoto/LuaLibBase.cpp:45:3: error: ‘luaL_openlib’ was not declared in this scope; did you mean ‘luaL_openlibs’?
   45 |   luaL_openlib(m_pL, i_libname.c_str(), i_reg, 0);
      |   ^~~~~~~~~~~~
      |   luaL_openlibs
```

Here some hints for replacement http://lua-users.org/wiki/CompatibilityWithLuaFive

**Comment by Nikekson** [comment]
Unless this can be fixed by retaining compatibility with scripted levels depending on older lua versions, use the vendored library instead

**Comment by winterheart** [comment]
Hello.
Can you provide example of these levels for testing?

**Comment by Nikekson** [comment]
Oh, I'm sorry, after looking into this more, it seems like only functions in lua's C api have been deprecated.
Fixing this should be as simple as:
```diff
@@ -42,7 +42,9 @@
   luaL_requiref(m_pL, LUA_TABLIBNAME, luaopen_table, 1);
 #endif
 
-  luaL_openlib(m_pL, i_libname.c_str(), i_reg, 0);
+  lua_newtable(m_pL);
+  luaL_register(m_pL, i_libname.c_str(), i_reg);
+  lua_setglobal(m_pL, i_libname.c_str());
 }
 
 LuaLibBase::~LuaLibBase() {
```
Does this work for you?
You can test it on any scripted level.

**Comment by Nikekson** [comment]
I've pushed a more proper fix in 518685d, let me know if that fixes it

**Comment by winterheart** [comment]
Looks good, tested on backported 0.6.0, compiles and runs fine.

BTW, `luaL_register` is also being deprecated in Lua 5.2, but this is another story I guess...

**Comment by Nikekson** [comment]
> BTW, luaL_register is also being deprecated in Lua 5.2

Fixed in 0a92ee4

**Comment by winterheart** [comment]
Looks good for me!
